src/templates/common/medallia/index.test.tsx

```diff
@@ -33,4 +98,81 @@ describe('MedalliaAssets Component', () => {
       'https://resource.digital.voice.va.gov/wdcvoice/5/onsite/embed.js'
     expect(scriptElement.getAttribute('data-src')).toEqual(expectedSrc)
   })
+  test('adds listener and calls recordEvent when triggered for Medallia event with UUID and Content', () => {
+    const mockEventDetail = {
+      ...mockEvent,
+      detail: {
+        Form_Type: 'Test Form',
+        Form_ID: '12345',
+        Feedback_UUID: 'abc-123',
+        Content: 'Test Content',
+      },
+    }
+    const addEventListenerSpy = jest.spyOn(window, 'addEventListener')
+
+    render(<MedalliaAssets />)
+
+    expect(addEventListenerSpy).toHaveBeenCalledWith(
+      'MDigital_Feedback_Submit',
+      expect.any(Function),
+      expect.objectContaining({ signal: expect.any(AbortSignal) })
+    )
+
+    // Simulate the event listener being triggered
+    const eventHandler = addEventListenerSpy.mock.calls.find(
+      (call): call is [string, (event: CustomEvent) => void] =>
+        call[0] === 'MDigital_Feedback_Submit'
+    )?.[1]
+    if (eventHandler) {
+      eventHandler(mockEventDetail)
+    }
```

Comment:
Whoa, this is fancy. Just to make sure I understand:
1. When we render the component, it fires off that new `useEffect()`
2. This `useEffect()` adds some event listeners
3. We're finding a specific event listener by hunting through the calls to our `addEventListener` spy for a specific one, then calling it with a mocked event, thereby simulating sending that actual event, which would normally trigger the listener if we _actually_ added it to `window`'s event listeners

Not sure I fully understand the need for that type guard, but I assume it's useful because of Reasons™. 😅

Reply:
Yes you are 100% correct in your understanding. The type guard is in place because the typescript linter wanted it 🤷  It all looked much simpler when I first added it.